docs/howto/partitions.rst

```diff
@@ -0,0 +1,193 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+.. Insert link below to new Snapcraft docs when this is merged and live:
+   https://github.com/canonical/snapcraft/issues/4857
+
+Applications implementing ``craft-application`` and its dependent suite of
+libraries can optionally make use of *partitions* when organizing the files in
+their output artifact.  Partitions are a generic concept, and may be used to
+implement a concrete feature in your application.  For instance, partitions may
+refer to:
```

Comment:
WDYT about "Partitions are used to organize files into different artifacts or artifact sections, and the actual layout of partitions are defined by the application. For instance, partitions can be used to implement:" or something similar?

Reply:
Fixed here: https://github.com/canonical/craft-application/pull/381